extension/basicauthextension/README.md

```diff
@@ -1,47 +1,58 @@
 # Basic Authenticator
 
-This extension implements `configauth.ServerAuthenticator` to authenticate clients using HTTP Basic Authentication. The authenticator type has to be set to `basicauth`.
+This extension implements both `configauth.ServerAuthenticator` and `configauth.ClientAuthenticator` to authenticate clients and servers using HTTP Basic Authentication. The authenticator type has to be set to `basicauth`.
 
-If authentication is successful `client.Info.Auth` will expose the following attributes:
+When used as ServerAuthenticator, if the authentication is successful `client.Info.Auth` will expose the following attributes:
 
 - `username`: The username of the authenticated user.
 - `raw`: Raw base64 encoded credentials.
 
+The configuration should specify only one instance of `basicauth` extension for either client or server authentication. 
+
+The following are the configuration options:
+
+- `htpasswd.file`:  The path to the htpasswd file.
+- `htpasswd.inline`: The htpasswd file inline content.
+- `htpasswd.client_auth`: Single username password combination in the form of `username:password` for client authentication.
+
+To configure the extension as a server authenticator, either one of `htpasswd.file` or `htpasswd.inline` has to be set. If both are configured, `htpasswd.inline` credentials take precedence.
+
+To configure the extension as a client authenticator, `htpasswd.client_auth` has to be set.
```

Comment:
Shouldn't it be like this according to the actual code?

```suggestion
- `client_auth`: Single username password combination in the form of `username:password` for client authentication.

To configure the extension as a server authenticator, either one of `htpasswd.file` or `htpasswd.inline` has to be set. If both are configured, `htpasswd.inline` credentials take precedence.

To configure the extension as a client authenticator, `client_auth` has to be set.
```

Reply:
Oh yes. Nice catch. Thanks for reviewing 😁